### PR TITLE
GH-15684 - redirect all logs for level ERROR and FATAL also to stderr

### PR DESCRIPTION
**Pull request by valenad1**
#15684

I configured our default logging to put ERROR and FATAL log levels also to stderr

### CODE REVIEW

**Comment by valenad1** [comment]
> Is it possible to add a test?

Its more difficult than I would expect. Only think that comes to my mind is to redirect System.err to different print stream. It doesn't work for LOG.error() but only if you call System.err.println() directly. Add different appender also doesn't work. I will try something tomorrow if something comes to my mind..

**Review by wendycwong — DISMISSED** [review]

**Comment by valenad1** [comment]
> Is it possible to add a test?

Done. I have created a new stage to test it in shell directly.

**Review by wendycwong — APPROVED** [review]

File: h2o-logging/impl-log4j2/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -4,10 +4,15 @@
...
         </Console>
+        <Console name="stderr" target="SYSTEM_ERR"> <!-- 2 -->
+            <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
+            <ThresholdFilter level="ERROR" onMatch="ACCEPT" onMismatch="DENY"/> <!-- 3 -->
+        </Console>
     </Appenders>
     <Loggers>
         <Root level="info">
             <AppenderRef ref="Console"/>
+            <AppenderRef ref="stderr"/>
```
Comment (mn-mikke) [inline h2o-logging/impl-log4j2/src/main/resources/log4j2.xml:15]: Wouldn't this causes that info messages will go t stderr as well?

Comment (valenad1) [reply h2o-logging/impl-log4j2/src/main/resources/log4j2.xml:15]: It wouldn't. Here is only declaration that this appender exists, the behavior of the appender is defined in upper lines

**Review by mn-mikke — DISMISSED** [review]
LGTM

**Review by mn-mikke — APPROVED** [review]
LGTM

**Pull request merged**